### PR TITLE
feat: Cart page

### PR DESCRIPTION
**Pull request by rutu14**
### Cart Page
- The cart page requires login and if the user is not logged in, message is displayed.
- On Product Page, add to cart button adds the product to the cart and the notification badge changes.
- The remove cart item, increase qty and decrease qty works fine.
- The cart summary part shows the amount as per the cart items.
- Issue arises that the summary doesn't refresh on remove cart item, increase qty and decrease qty, so move to product page and go back to cart page. The summary is updated. [ Working on this point ]

[Click to view files](https://github.com/rutu14/Ecommerce-React/pull/6/commits/ee5f84c8de4e992d578def1d12c6c32a99ed0424)

### CODE REVIEW

File: src/reducers/cartReducer.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                loader: false,
+                productAdded: true
+            });
+        case "CART_SUCCESS_MATCH_UNDO":
+            return (state = {
+                ...state,
+                loader: false,
+                productAdded: false
+            });
+        case "CART_SUCCESS_MATCH_UNDO":
```
Comment (amartya-rs) [inline src/reducers/cartReducer.js]: Repeated CART_SUCCESS_MATCH_UNDO case. Remove one.

File: src/pages/Cart-Page/CartPage.jsx [hunk 2]
```diff
@@ -1,6 +1,54 @@
...
+    useEffect(()=>{
+        getCart()
+        subTotal()
+    },[ ])
+
+    const [ priceCal, setPriceCal ] = useState(0.00)
+
+    const subTotal = () => {
+        cartState.cartInfo.map(( cartValue ) => {
+            console.log(cartValue.price)
```
Comment (amartya-rs) [inline src/pages/Cart-Page/CartPage.jsx]: remove console log

File: src/context/cartContext.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        }
+    }
+
+    const deleteCart = async ( productId ) => {
+        try {
+            dispatch({ type: "CART_REQUEST" })
+            const token = localStorage.getItem("token");
+            const config = { headers: { 'authorization': token } };
+            const { data } = await axios.delete( `/api/user/cart/${productId}`, config );
+            console.log(data)
```
Comment (amartya-rs) [inline src/context/cartContext.js]: remove console log

**Review by amartya-rs — APPROVED** [review]
Code is well structured. Good to merge.

File: src/reducers/cartReducer.js (hunk 1 above)
Comment (rutu14) [reply src/reducers/cartReducer.js]: Thanks for the reminder.

File: src/pages/Cart-Page/CartPage.jsx (hunk 2 above)
Comment (rutu14) [reply src/pages/Cart-Page/CartPage.jsx]: Thanks for the reminder.

File: src/context/cartContext.js (hunk 3 above)
Comment (rutu14) [reply src/context/cartContext.js]: Thanks for the reminder.

**Comment by santhoshpemmaka** [comment]
No suggestion from my side. The code looks good and clean. 💯 . Good to merge 👍

**Review by santhoshpemmaka — APPROVED** [review]

**Pull request merged**